### PR TITLE
Adds a proxy that attests the client side

### PR DESCRIPTION
**Pull request by Ruteri**

### CODE REVIEW

**Comment by metachris** [comment]
https://github.com/edgelesssys/constellation/blob/main/internal/atls/README.md#server-side-verification

**Review by metachris — APPROVED** [review]

File: cmd/proxy-client/main.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		Usage: "log debug messages",
+	},
+}
+
+func main() {
+	app := &cli.App{
+		Name:   "proxy-client",
+		Usage:  "Serve API, and metrics",
+		Flags:  flags,
+		Action: client_side_tls_termination,
```
Comment (metachris) [inline cmd/proxy-client/main.go:56]:
```suggestion
		Action: run_client,
```

File: cmd/proxy-server/main.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		Usage: "log debug messages",
+	},
+}
+
+func main() {
+	app := &cli.App{
+		Name:   "proxy-server",
+		Usage:  "Serve API, and metrics",
+		Flags:  flags,
+		Action: server_side_tls_termination,
```
Comment (metachris) [inline cmd/proxy-server/main.go:61]:
```suggestion
		Action: run_server,
```

File: proxy/atls_config.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+)
+
+type AttestationType string
+
+const (
+	AttestationNone     AttestationType = "none"
+	AttestationAzureTDX AttestationType = "azure-tdx"
+)
+
+const AvailableAttestationTypes string = "none, azure-tdx"
```
[the remark this answers is not in the document]
Comment (Ruteri) [reply proxy/atls_config.go:23]: It's for sure not as ugly as concatenating the consts!

**Pull request merged**